src/zcl_abapgit_tadir.clas.abap

```diff
@@ -277,4 +292,61 @@ CLASS ZCL_ABAPGIT_TADIR IMPLEMENTATION.
     ENDIF.
 
   ENDMETHOD.
+
+
+  METHOD mark_inactive_objects.
+
+    DATA: messages TYPE STANDARD TABLE OF sprot_u WITH DEFAULT KEY.
+
+    mark_inactive_special_objects( CHANGING c_check_tadirs = c_check_tadirs ).
+
+    CALL FUNCTION 'RS_INACTIVE_OBJECTS_WARNING'
+      EXPORTING
+        suppress_protocol         = abap_false
+        with_program_includes     = abap_false
+        suppress_dictionary_check = abap_false
+        phased_activation         = abap_false
+      TABLES
+        p_e071                    = c_e071_tadirs
+        p_xmsg                    = messages.
+
+    LOOP AT messages ASSIGNING FIELD-SYMBOL(<message>).
+      READ TABLE c_check_tadirs ASSIGNING FIELD-SYMBOL(<check_tadir>)
+                                WITH KEY object = <message>-var1 obj_name = <message>-var2.
+      IF sy-subrc <> 0.
+        READ TABLE c_check_tadirs ASSIGNING <check_tadir> WITH KEY obj_name = <message>-var2.
+        IF sy-subrc <> 0.
+          LOOP AT c_check_tadirs ASSIGNING <check_tadir>.
+            IF <message>-var2 CP |{ <check_tadir>-obj_name }=*|. "some more checks needed?
+              EXIT. "LOOP
+            ENDIF.
+          ENDLOOP.
+        ENDIF.
+      ENDIF.
+
+      IF <check_tadir> IS ASSIGNED.
+        <check_tadir>-inactive = abap_true.
+      ENDIF.
+      UNASSIGN <check_tadir>.
+    ENDLOOP.
+
+  ENDMETHOD.
+
+
+  METHOD mark_inactive_special_objects.
+
+    DATA: ssfo_formname TYPE tdsfname.
+
+* SSFO is special => the only one???
+    LOOP AT c_check_tadirs ASSIGNING FIELD-SYMBOL(<check_tadir>) WHERE object = 'SSFO'.
```

Comment:
TADIR usage is quite messed up :) so I would expect more special objects

I suggest moving the logic for determining if the object is inactive to `INTERFACE zif_abapgit_object`, the `is_inactive` method belongs to the same category of methods as following which is already part of the interface,

```abap
    exists
      RETURNING VALUE(rv_bool) TYPE abap_bool
      RAISING   zcx_abapgit_exception,
    is_locked
      RETURNING VALUE(rv_is_locked) type abap_bool
      RAISING
        zcx_abapgit_exception,
```

Also, by moving the is_inactive determination to object serialization, will make it possible to move the calculation to parallel task when https://github.com/larshp/abapGit/issues/1714 is implemented